org.eclipse.ua.tests.doc/src/org/eclipse/ua/tests/doc/internal/linkchecker/ApiDocTest.java

```diff
@@ -276,31 +277,17 @@ private static InputSource getExtensionPointSchemaSource(IExtensionPoint extensi
 	private static void checkPackages(Set<String> packageIds, StringBuilder problems) {
 		Set<String> exportedPackageIds = new TreeSet<>();
 
-		exportedPackageIds.add("org.eclipse.core.runtime.adaptor"); // not exported, but makes sense to document since accessible from outside of OSGi framework
-		exportedPackageIds.add("org.eclipse.swt.ole.win32"); // somehow missing from State#getExportedPackages(), maybe because it's declared in the fragment only
+		exportedPackageIds.add("org.eclipse.core.runtime.adaptor"); // not exported, but makes sense to document since
+																	// accessible from outside of OSGi framework
+		exportedPackageIds.add("org.eclipse.swt.ole.win32"); // somehow missing from State#getExportedPackages(), maybe
+																// because it's declared in the fragment only
 		BundleContext context = FrameworkUtil.getBundle(ApiDocTest.class).getBundleContext();
 		ServiceReference<PlatformAdmin> platformAdminReference = context.getServiceReference(PlatformAdmin.class);
 		PlatformAdmin service = context.getService(platformAdminReference);
 		State state = service.getState(false);
-		ExportPackageDescription[] exportedPackages = state.getExportedPackages();
-		for (ExportPackageDescription exportPackageDescription : exportedPackages) {
-			String name = exportPackageDescription.getName();
-			if (!NON_API_PACKAGES.matcher(name).matches()) {
-				if (Boolean.TRUE.equals(exportPackageDescription.getDirective("x-internal"))
-						|| exportPackageDescription.getDirective("x-friends") != null) {
```

Comment:
So we check for all docs now, including internal/friends/NON_API_PACKAGES ?
I don't think it's too bad, just want to make sure it's the actual intent/fix.

Reply:
Actually it is the opposite. These package names are removed from packageIds and it's checked whether there is anything left. IMHO, it's fine to have x-internal/x-friend packages in the javadoc and this test shouldn't fail in such a case.